src/main/scala/tech/cryptonomic/conseil/michelson/parser/JsonParser.scala

```diff
@@ -0,0 +1,94 @@
+package tech.cryptonomic.conseil.michelson.parser
+
+import io.circe.parser.decode
+import cats.syntax.functor._
+import io.circe._
+import io.circe.generic.auto._
+import tech.cryptonomic.conseil.michelson.dto._
+
+class JsonParser {
+
+  sealed trait JsonSection
+
+  case class JsonTypeSection(prim: String, args: Option[Seq[JsonTypeSection]]) extends JsonSection {
+    def toMichelsonType: MichelsonType = MichelsonType(prim, args.getOrElse(Seq()).map(_.toMichelsonType))
+  }
+
+  case class ExpressionSection(prim: String, args: Seq[Seq[JsonInstruction]]) extends JsonSection {
+    def toMichelsonExpression = MichelsonCode(args.flatten.map(_.toMichelsonInstruction))
+  }
+
+  sealed trait JsonInstruction {
+    def toMichelsonInstruction: MichelsonInstruction
+  }
+
+  case class JsonSimpleInstruction(prim: String, args: Option[Seq[JsonTypeSection]] = None) extends JsonInstruction {
+    override def toMichelsonInstruction = MichelsonSimpleInstruction(prim, args.flatMap(_.headOption.map(_.toMichelsonType)))
+  }
+
+  case class JsonComplexInstruction(prim: String, args: Seq[Seq[JsonInstruction]]) extends JsonInstruction {
+    override def toMichelsonInstruction = MichelsonComplexInstruction(prim, MichelsonInstructionSequence(args.flatten.map(_.toMichelsonInstruction)))
+  }
+
+  case class JsonInstructionSequence(instructions: Seq[JsonInstruction]) extends JsonInstruction {
+    override def toMichelsonInstruction = MichelsonInstructionSequence(instructions.map(_.toMichelsonInstruction))
+  }
+
+  class ParserError(message: String) extends Throwable(message)
+
+  type Result[T] = Either[Throwable, T]
+
+  case class JsonDocument(code: Seq[JsonSection]) {
+    def toMichelsonSchema: Result[MichelsonSchema] = {
+      for {
+        parameter <- extractType("parameter")
+        storage <- extractType("storage")
+        code <- extractExpression("code")
+      } yield MichelsonSchema(parameter, storage, code)
+    }
+
+    private def extractType(parameter: String): Result[MichelsonType] = {
```

Comment:
Maybe we could find a more expressive name than `parameter: String`?
It can get confusing...